src/components/dropdown.component.js

```diff
@@ -1,43 +1,93 @@
-import $ from 'jquery';
-
-export default class DropdownComponent {
-    constructor(query) {
-        this.query = query;
-
-        this.btn = $(`${query} .dropdown-btn`);
-        this.items = $(`${query} .dropdown-item`);
-        this.list = $(`${query} .dropdown-list`);
-        this.selectedItem = $(`${query} .selected`);
-
-        this.addDropdownClickEventListner();
-        this.addItemClickEventListner();
-    }
-
-    addDropdownClickEventListner() {
-        this.btn.on('click', () => {
-            this.list.toggle();
-        });
-    }
-
-    addItemClickEventListner() {
-        this.items.on('click', (event) => {
-            this.unselectItem();
-            this.selectItem(event.currentTarget);
-            this.list.toggle();
-        });
-    }
-
-    updateDropdown() {
-        this.btn.html(this.selectedItem.text());
-    }
-
-    unselectItem() {
-        this.items.removeClass('selected');
-    }
-
-    selectItem(item) {
-        this.selectedItem = $(item);
-        this.selectedItem.addClass('selected');
-        this.updateDropdown();
-    }
-}
+import $ from 'jquery';
+
+export default class DropdownComponent {
+    constructor(query, required = true) {
+        this.query = query;
+        this.required = required;
+
+        this.container = $(`${query} .dropdown-container`);
+        this.btn = $(`${query} .dropdown-btn`);
+        this.list = $(`${query} .dropdown-list`);
+        this.items = $(`${query} .dropdown-item`);
+        this.selectedItem = $(`${query} .selected`);

```

Comment:
I believe that this is not a component state property and would be something private, in which case the _ can be used to indicate that it should be used within the class.
Ex: '_selectedItem'
![image](https://user-images.githubusercontent.com/23269934/162783448-a0e2b627-3ccd-4031-a22c-050389c38a11.png)

Reply:
Done